### PR TITLE
Optimizations for printing disassembly

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* Stops rizin from reloading arch profiles every time you move in disassembly
* Optimizes calculating current row/column
* Caches console output buffer size to avoid expensive syscall

**Test plan**

Tests pass, disassembly works

**Closing issues**

None

### CODE REVIEW

File: librz/include/rz_cons.h [hunk 1]
```diff
@@ -487,6 +487,11 @@ typedef struct rz_cons_context_t {
 	int color_mode;
 	RzConsPalette cpal;
 	RzConsPrintablePalette pal;
+	
+	// Memoized last calculated row/column inside buffer
```
Comment (wargio) [inline librz/include/rz_cons.h:491]: Save the last calculated row/column...

File: librz/include/rz_util/rz_print.h [hunk 2]
```diff
@@ -120,6 +120,8 @@ typedef struct rz_print_t {
 	// represents the first not-visible offset on the screen
 	// (only when in visual disasm mode)
 	ut64 screen_bounds;
+	// Memoized current row number to calculate screen_bounds
```
Comment (wargio) [inline librz/include/rz_util/rz_print.h:123]: Save current row number ...

File: librz/util/print.c [hunk 3]
```diff
@@ -466,8 +466,6 @@ RZ_API void rz_print_hexii(RzPrint *rp, ut64 addr, const ut8 *buf, int len, int
 /* set screen_bounds to addr if the cursor is not visible on the screen anymore.
```
Comment (XVilka) [inline librz/util/print.c]: If you change this function - please convert the description to Doxygen.

**Review by XVilka — CHANGES_REQUESTED** [review]
It broke some of the output in tests:
```

[XX] db/cmd/cmd_pd asm.cmt.esil
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -F prg -Qc 'e asm.cmt.esil=1
e asm.cmt.right=0
pd 3 @ 0x80e
?e --
e asm.cmt.right=1
pd 3 @ 0x80e
' bins/prg/t01.prg
-- stdout
--- expected
+++ actual
@@ -1,10 +1,7 @@
-            ; 0x41,a,=,$z,Z,:=,7,$s,N,:=
             0x0000080e      lda #0x41
-            ; 0xff,x,=,$z,Z,:=,7,$s,N,:=
             0x00000810      ldx #0xff
-            ; a,x,0x03ff,+,=[1]
             0x00000812      sta 0x03ff,x
 --
-            0x0000080e      lda   #0x41                                ; 0x41,a,=,$z,Z,:=,7,$s,N,:=
-            0x00000810      ldx   #0xff                                ; 0xff,x,=,$z,Z,:=,7,$s,N,:=
-            0x00000812      sta   0x03ff,x                             ; a,x,0x03ff,+,=[1]
+            0x0000080e      lda   #0x41
+            0x00000810      ldx   #0xff
+            0x00000812      sta   0x03ff,x
-- stderr
WARNING: No calling convention defined for this file, analysis may be inaccurate.
[XX] db/cmd/cmd_pd asm.cmt.il
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -F prg -Qc 'e asm.cmt.il=1
e asm.cmt.right=0
pd 3 @ 0x80e
?e --
e asm.cmt.right=1
pd 3 @ 0x80e
' bins/prg/t01.prg
-- stdout
--- expected
+++ actual
@@ -1,10 +1,7 @@
-            ; (seq (set a (bv 8 0x41)) (set Z (is_zero (var a))) (set N (msb (var a))))
             0x0000080e      lda #0x41
-            ; (seq (set x (bv 8 0xff)) (set Z (is_zero (var x))) (set N (msb (var x))))
             0x00000810      ldx #0xff
-            ; (store 0 (+ (bv 16 0x3ff) (cast 16 false (var x))) (var a))
             0x00000812      sta 0x03ff,x
 --
-            0x0000080e      lda   #0x41                                ; (seq (set a (bv 8 0x41)) (set Z (is_zero (var a))) (set N (msb (var a))))
-            0x00000810      ldx   #0xff                                ; (seq (set x (bv 8 0xff)) (set Z (is_zero (var x))) (set N (msb (var x))))
-            0x00000812      sta   0x03ff,x                             ; (store 0 (+ (bv 16 0x3ff) (cast 16 false (var x))) (var a))
+            0x0000080e      lda   #0x41
+            0x00000810      ldx   #0xff
+            0x00000812      sta   0x03ff,x
-- stderr
WARNING: No calling convention defined for this file, analysis may be inaccurate.
[XX] db/analysis/avr avr DISasm: IO-Ports
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'pd 13 @ 0x00000282' bins/firmware/arduino_avr.bin
-- stdout
--- expected
+++ actual
@@ -1,13 +1,13 @@
-            0x00000282      in    r24, 0x12                            ; IO Output pins/pullups address for port D
+            0x00000282      in    r24, 0x12
             0x00000284      subi  r24, 0x80
-            0x00000286      out   0x12, r24                            ; IO Output pins/pullups address for port D
+            0x00000286      out   0x12, r24
             0x00000288      ret
             0x0000028a      ldi   r22, 0x7d
             0x0000028c      ldi   r23, 0x01
             0x0000028e      ldi   r24, 0x2f
             0x00000290      ldi   r25, 0x02
             0x00000292      call  0x8e6
-            0x00000296      in    r24, 0x18                            ; IO Output pins/pullups address for port B
+            0x00000296      in    r24, 0x18
             0x00000298      subi  r24, 0x80
-            0x0000029a      out   0x18, r24                            ; IO Output pins/pullups address for port B
+            0x0000029a      out   0x18, r24
             0x0000029c      ret
```
https://github.com/rizinorg/rizin/runs/6841381548?check_suite_focus=true#step:12:280

File: librz/core/disasm.c [hunk 4]
```diff
@@ -17,6 +17,8 @@
 #define COLOR_CONST(ds, color) ((ds)->show_color ? Color_##color : "")
 #define COLOR_RESET(ds)        COLOR_CONST(ds, RESET)
 
+#define DS_ANALYSIS_OP_MASK (RZ_ANALYSIS_OP_MASK_BASIC | RZ_ANALYSIS_OP_MASK_ESIL | RZ_ANALYSIS_OP_MASK_VAL | (ds->show_cmt_il ? RZ_ANALYSIS_OP_MASK_IL : 0))
```
Comment (XVilka) [inline librz/core/disasm.c]: Please split this long line.

**Review by XVilka — CHANGES_REQUESTED** [review]
Looks good from the code PoV but there are two tests broken:
```

[XX] db/cmd/cmd_pd asm.cmt.esil
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -F prg -Qc 'e asm.cmt.esil=1
e asm.cmt.right=0
pd 3 @ 0x80e
?e --
e asm.cmt.right=1
pd 3 @ 0x80e
' bins/prg/t01.prg
-- stdout
--- expected
+++ actual
@@ -1,10 +1,7 @@
-            ; 0x41,a,=,$z,Z,:=,7,$s,N,:=
             0x0000080e      lda #0x41
-            ; 0xff,x,=,$z,Z,:=,7,$s,N,:=
             0x00000810      ldx #0xff
-            ; a,x,0x03ff,+,=[1]
             0x00000812      sta 0x03ff,x
 --
-            0x0000080e      lda   #0x41                                ; 0x41,a,=,$z,Z,:=,7,$s,N,:=
-            0x00000810      ldx   #0xff                                ; 0xff,x,=,$z,Z,:=,7,$s,N,:=
-            0x00000812      sta   0x03ff,x                             ; a,x,0x03ff,+,=[1]
+            0x0000080e      lda   #0x41
+            0x00000810      ldx   #0xff
+            0x00000812      sta   0x03ff,x
-- stderr
WARNING: No calling convention defined for this file, analysis may be inaccurate.

[XX] db/analysis/arm arm-or-thumb visual bug
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e scr.interactive=1
e scr.null=1
Vprdfq
e scr.null=0
afi~^size[1]
' bins/mach0/arm-or-thumb
-- stdout
--- expected
+++ actual
@@ -1,1 +1,0 @@
-32
-- exit status: -1
[**]                        
```

File: librz/include/rz_arch.h [hunk 5]
```diff
@@ -39,6 +39,7 @@ typedef struct rz_platform_item_t {
 
 typedef struct rz_platform_target_t {
 	HtUP /* <ut64 , RzArchPlatformItem> */ *platforms;
+	char *path;
```
Comment (ret2libc) [inline librz/include/rz_arch.h]: doxygen doc of what this represents

File: librz/cons/cons.c [hunk 6]
```diff
@@ -1260,38 +1270,54 @@ now the console color is reset with each \n (same stuff do it here but in correc
...
 
-/* return the aproximated x,y of cursor before flushing */
-// XXX this function is a huge bottleneck
+/**
+ * \brief Calculates the aproximated x,y coordinates of the cursor before flushing
+ * \param[out] rows Row number of the cursor
+ * \return Column number of the cursor
+ */
 RZ_API int rz_cons_get_cursor(int *rows) {
-	int i, col = 0;
```
Comment (XVilka) [inline librz/cons/cons.c:1266]: I recommend also adding `rz_return_val_if_fail(rows, 0)` and `RZ_NONNULL`

**Review by XVilka — APPROVED** [review]
Looks good overall, please address these small comments.

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**